File: 代码随想录训练营/day23(108).cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
#include "DataStruct.h"
#include "vector"
using namespace std;
class Solution {
public:
    TreeNode* travel(vector<int>& nums, int left, int right){
        if (left > right) return nullptr;
        int mid = left + ((right - left) >> 1);
        TreeNode* root = new TreeNode(nums[mid]);
        root -> left = travel(nums, left, mid - 1);
        root -> right = travel(nums, mid + 1, right);
        return root;
    }


    TreeNode* sortedArrayToBST(vector<int>& nums) {
        TreeNode* res = travel(nums, 0, nums.size() - 1);
        return res;
    }
};

File: 代码随想录训练营/day23(108).py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> Optional[TreeNode]:
        root = self.travel(nums, 0, len(nums) - 1)
        return root


    def travel(self, nums, left, right):
        if left > right: return None
        mid = left + ((right - left) // 2)
        root = TreeNode(nums[mid])
        root.left = self.travel(nums, left, mid - 1)
        root.right = self.travel(nums, mid + 1, right)
        return root


